snapcraft/extensions/kde_neon_6.py

```diff
@@ -0,0 +1,272 @@
+            "assumes": ["snapd2.58.3"],  # for 'snapctl is-connected'
+            "compression": "lzo",
+            "plugs": {
+                "desktop": {"mount-host-font-cache": False},
+                "icon-themes": {
+                    "interface": "content",
+                    "target": "$SNAP/data-dir/icons",
+                    "default-provider": "gtk-common-themes",
+                },
+                "sound-themes": {
+                    "interface": "content",
+                    "target": "$SNAP/data-dir/sounds",
+                    "default-provider": "gtk-common-themes",
+                },
+                platform_qt6_snap: {
+                    "content": content_qt6_snap,
+                    "interface": "content",
+                    "default-provider": platform_qt6_snap,
+                    "target": "$SNAP/qt6",
+                },
+                platform_kf6_snap: {
+                    "content": content_kf6_snap,
+                    "interface": "content",
+                    "default-provider": platform_kf6_snap,
+                    "target": "$SNAP/kf6",
+                },
+            },
+            "environment": {"SNAP_DESKTOP_RUNTIME": "$SNAP/kf6"},
+            "hooks": {
+                "configure": {
+                    "plugs": ["desktop"],
+                    "command-chain": ["snap/command-chain/hooks-configure-desktop"],
+                }
+            },
+            "layout": {"/usr/share/X11": {"symlink": "$SNAP/kf6/usr/share/X11"}},
+        }
+
+    @overrides
+    def get_part_snippet(self, *, plugin_name: str) -> Dict[str, Any]:
+        qt6_sdk_snap = self.kde_snaps.qt6_sdk_snap
+        kf6_sdk_snap = self.kde_snaps.kf6_sdk_snap
+        cmake_args = self.ext_info.cmake_args
+
+        return {
+            "build-environment": [
+                {
+                    "PATH": prepend_to_env(
+                        "PATH",
+                        [
+                            f"/snap/{qt6_sdk_snap}/current/"
+                            f"usr/bin:/snap/{kf6_sdk_snap}/"
+                            f"current/usr/bin"
+                        ],
+                    ),
+                },
+                {
+                    "XDG_DATA_DIRS": prepend_to_env(
+                        "XDG_DATA_DIRS",
+                        [
+                            f"$CRAFT_STAGE/usr/share:"
+                            f"/snap/{qt6_sdk_snap}"
+                            f"/current/usr/share:"
+                            f"/snap/{kf6_sdk_snap}"
+                            f"/current/usr/share",
+                            "/usr/share",
+                        ],
+                    ),
+                },
+                {
+                    "SNAPCRAFT_CMAKE_ARGS": prepend_to_env(
+                        "SNAPCRAFT_CMAKE_ARGS",
+                        [
+                            cmake_args,
+                        ],
+                    ),
+                },
```

Comment:
Looking at the [ark snap](https://invent.kde.org/scarlettmoore/ark/-/blob/master/snapcraft.yaml?ref_type=heads), would these additions make sense?

They work for me locally and still let me build an apparently-functional snap of ark, but I'm not sure what downsides there may be:

```suggestion
                },
                {
                    "XDG_CONFIG_HOME": prepend_to_env(
                        "XDG_CONFIG_HOME",
                        [
                            "$CRAFT_STAGE/etc/xdg",
                            f"/snap/{qt6_sdk_snap}/current/etc/xdg",
                            f"/snap/{kf6_sdk_snap}/current/etc/xdg",
                            "/etc/xdg",
                        ],
                    ),
                },
                {
                    "LD_LIBRARY_PATH": prepend_to_env(
                        "LD_LIBRARY_PATH",
                        [
                            f"/snap/{qt6_sdk_snap}/current/usr/lib/${{CRAFT_ARCH_TRIPLET_BUILD_FOR}}",
                            f"/snap/{kf6_sdk_snap}/current/usr/lib/${{CRAFT_ARCH_TRIPLET_BUILD_FOR}}",
                            f"/snap/{qt6_sdk_snap}/current/usr/lib",
                            f"/snap/{kf6_sdk_snap}/current/usr/lib",
                            "${CRAFT_STAGE}/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}"
                            "${CRAFT_STAGE}/usr/lib",
                            "${CRAFT_STAGE}/lib/",
                        ],
                    ),
                },
                {
                    "CMAKE_PREFIX_PATH": prepend_to_env(
                        "CMAKE_PREFIX_PATH",
                        [
                            "$CRAFT_STAGE",
                            f"/snap/{qt6_sdk_snap}/current",
                            f"/snap/{kf6_sdk_snap}/current",
                            "/usr",
                        ],
                        separator=";",
                    ),
                },
                {
                    "CMAKE_FIND_ROOT_PATH": prepend_to_env(
                        "CMAKE_FIND_ROOT_PATH",
                        [
                            "$CRAFT_STAGE",
                            f"/snap/{qt6_sdk_snap}/current",
                            f"/snap/{kf6_sdk_snap}/current",
                            "/usr",
                        ],
                        separator=";",
                    ),
                },
```